operators/audio/audio_decoder.hpp

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "ocos.h"
+
+#include <list>
+#include <map>
+#define DR_FLAC_IMPLEMENTATION
+#include "dr_flac.h"
+#define DR_MP3_IMPLEMENTATION 1
+#define DR_MP3_FLOAT_OUTPUT 1
+#include "dr_mp3.h"
+#define DR_WAV_IMPLEMENTATION
+#include "dr_wav.h"
+
+#include <gsl/util>
+#include "string_utils.h"
+#include "string_tensor.h"
+
+struct KernelAudioDecoder : public BaseKernel {
+ public:
+  KernelAudioDecoder(const OrtApi& api, const OrtKernelInfo& info) : BaseKernel(api, info) {
+  }
+
+  enum class AudioStreamType {
+    kDefault = 0,
+    kWAV,
+    kMP3,
+    kFLAC
+  };
+
+  AudioStreamType ReadStreamFormat(OrtKernelContext* context, const uint8_t* p_data) {
+    const std::map<std::string, AudioStreamType> format_mapping = {
```

Comment:
nit: could be static
```suggestion
    static const std::map<std::string, AudioStreamType> format_mapping = {
```